src/modules/performance_collector/db.py

```diff
@@ -0,0 +1,168 @@
+import sqlite3
+from contextlib import contextmanager
+from typing import Optional
+
+from src import variables
+from src.modules.performance_collector.codec import ProposalDuty, SyncDuty, EpochDataCodec, AttDutyMisses
+from src.types import EpochNumber
+from src.utils.range import sequence
+
+
+class DutiesDB:
+    def __init__(self, path: str):
+        self._path = path
+        self._conn = sqlite3.connect(
+            self._path, check_same_thread=False, timeout=variables.PERFORMANCE_COLLECTOR_DB_CONNECTION_TIMEOUT
+        )
```

Comment:
What if we have an optional `_conn` field that is opened via a separate method such as connect and reused in the`connection` context manager? The context manager checks whether a connection has been established and uses it, or opens a new one otherwise. Thus, clients that prefer to keep the connection open can use the `connect` method and close it eventually, while other clients can use the `connection` context manager. I would also consider using the latter only. I think it is cheap to open a connection, do whatever is needed and close it afterwards.